modules/core/include/opencv2/core/bindings_utils.hpp

```diff
@@ -81,6 +81,9 @@ AsyncArray testAsyncException()
     return p.getArrayResult();
 }
 
+namespace fs {
+    CV_EXPORTS_W cv::String getCacheDirectory(const char *sub_directory_name);
```

Comment:
Aliases should work:
```
CV_EXPORTS_AS(getCacheDirectory) cv::String getCacheDirectory_bindings(const char *sub_directory_name, ...);
```

Also repair configuration parameter. It is necessary to keep OpenCV configurable.

---

IMHO, It make sense to hide some details and provide more specialized wrapper.

In `core/utility.hpp` add:
```
CV_EXPORTS_W cv::String getCacheDirectoryForDownloads(void);

```

with implementation (to avoid details in .py files):
```
{
    return cv::utils::fs::getCacheDirectory("downloads", "OPENCV_DOWNLOADS_CACHE_DIR");
}
```

Reply:
> In `core/utility.hpp` add:

I don't think it is a good idea to put function declaration and implementation there, because it would require including ```core/filesystem.hpp``` there. So I left it inside ```binding_utils```, which is in my opinion a more suitable place for a binding wrapper.